### PR TITLE
Part1&2

### PR DESCRIPTION
**Pull request by anisess10**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I am implementing and evaluating a KNN model by tuning hyperparameters, splitting the dataset, and calculating the model's performance.

## What did you learn from the changes you have made?
I learned how to optimize the KNN model using grid search, ensure reproducibility through random seed settings, and evaluate performance using accuracy metrics.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I encountered challenges with missing or improperly defined variables (e.g., response variable) and syntax errors. These were resolved by debugging, carefully reviewing the code.

## How were these changes tested?
The changes were tested by validating its performance using accuracy_score on the test set.

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by KasraVakiloroayaei — CHANGES_REQUESTED** [review]
A1: Incomplete 

The branch name for your repo should be: assignment-1
Q2part(ii) additionally since it is a categorical variable, we do not standardize it 
Q3-  It appears that you have the wrong identified the wrong value for your hyperparameter k, nearest Neighbors. Typically a k of 1 leads to model that is overfit. I would check to see if you have used the standardized data 
Q4 calculate accuracy again after changes above are made

**Comment by anisess10** [comment]
> A1: Incomplete
> 
> The branch name for your repo should be: assignment-1 Q2part(ii) additionally since it is a categorical variable, we do not standardize it Q3- It appears that you have the wrong identified the wrong value for your hyperparameter k, nearest Neighbors. Typically a k of 1 leads to model that is overfit. I would check to see if you have used the standardized data Q4 calculate accuracy again after changes above are made. 

@Melokazz Thank you for you feedback. As per the attached screenshot, my branch name is assignment-1 already. Could you please clarify the issue here? 
I've updated Q2 and reinitialized the model and cross-validation to fix the K issue that was resulting to 1. 
Please let me know if you have any other comments. 
Thanks!
Anis
Screenshot 2025-01-18 163107

**Review by KasraVakiloroayaei — APPROVED** [review]
A1: Complete 

Thank you for making the changes
